pkg/registry/core/service/strategy.go

```diff
@@ -263,10 +263,7 @@ func dropTypeDependentFields(newSvc *api.Service, oldSvc *api.Service) {
 }
 
 func needsClusterIP(svc *api.Service) bool {
-	if svc.Spec.Type == api.ServiceTypeExternalName {
-		return false
-	}
-	return true
+	return svc.Spec.Type != api.ServiceTypeExternalName
```

Comment:
not sure we can call that code optimization, imho it just makes the code harder to read